src/EPR.Accreditation.Facade/Services/AccreditationService.cs

```diff
@@ -339,5 +342,45 @@ public async Task<AddressDto> GetLegalDocumentsAddress(Guid id)
                 id,
                 accreditation);
         }
+
+        /// <summary>
+        /// Update the referernce number.
+        /// </summary>
+        /// <param name="id">The accrediation id.</param>
+        /// <returns></returns>
+        public async Task UpdateReferenceNumber(
+            Guid id)
+        {
+            var randomNumber = await _httpAccreditationService.GetRandomNumber(12);
+            var accreditation = new Common.Dtos.Accreditation
+            {
+                ReferenceNumber = randomNumber
+            };
+            
+            await _httpAccreditationService.UpdateAccreditation(id, accreditation);
+        }
+
+        /// <summary>
+        /// Returns the accrediation object.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public async Task <Accreditation> GetAccrediation(Guid id)
+        {
+            return await _httpAccreditationService.GetAccreditation(id);
+        }
+
+
+        /// <summary>
+        /// Builds a random string.
+        /// </summary>
+        /// <param name="length">Lenght of desired string.</param>
+        /// <returns>The random string.</returns>
+        private string RandomString(int length)
```

Comment:
I don't think this should be here any longer should it?

Reply:
Done.